source/server/admin/admin.cc

```diff
@@ -240,7 +240,7 @@ bool AdminImpl::createNetworkFilterChain(Network::Connection& connection,
   connection.addReadFilter(Network::ReadFilterSharedPtr{new Http::ConnectionManagerImpl(
       *this, server_.drainManager(), server_.api().randomGenerator(), server_.httpContext(),
       server_.runtime(), server_.localInfo(), server_.clusterManager(), null_overload_manager_,
-      server_.timeSource())});
+      server_.timeSource(), /*use_proactive_draining=*/false)});
```

Comment:
Disabling of draining on the admin interface means that we need to keep 2 code paths through the HCM indefinitely.  Is this divergence in behavior needed?

Reply:
It is at least needed for now as the admin interfaces assume they can always open new connections after entering a draining state.

I don't fully understand the use-cases for draining the admin interfaces. I would assume if this is the behavior that we want, that we don't need to drain these interfaces at all. If that were to be the case, then we could simply pass a new drain-manager instance (not connected to the server drain manager) that would never initiate draining.

Reply:
I've added a TODO. I can follow-up in another PR to ensure that the admin server does not drain.